### ISSUE TITLE
Transpiling error for `let` value assignment in loops Babel@5.5.5

### ISSUE THREAD

**Issue by mikeybox** [babel/babel#1707]
Applies to babel-core 5.5.5
When declaring variables in loop assignments there can be unexpected behaviour when a variable of the same name exists higher in scope. The loop iterator variable is incorrectly assigned as being the variable higher in the scope. Please see bellow examples. 

#### Example 1

```javascript
let i = 0;
for (let i = 0; i < 20; i++) {
	doSomethingElse();
}
```
*Result*

```javascript
let i = 0;
for (var _i = 0; _i < 20; _i++) {
	doSomethingElse();
}
```

#### Example 2

```javascript
var i = 0;
for (let i = 0; i < 20; i += 2) {
	doSomethingElse();
}
```
*Unexpected Result*

```javascript
var i = 0;
for (var _i = 0; _i < 20; i += 2) {
	doSomethingElse();
}
```

#### Example 3

```javascript
let i = 2;
function doSomething(){
	for (let i = 0; i < 20; i += 2) {
		doSomethingElse();
	}
}
```
*Unexpected Result*

```javascript
var i = 2;
function doSomething() {
	for (var _i = 0; _i < 20; i += 2) {
		doSomethingElse();
	}
}
```

**Comment by sebmck** [comment]
Regression due to 5aef7afade0f7c8eec9737ea89a63a89332e0b79.

**Comment by sebmck** [comment]
Fixed as of 5.5.6, thanks!

**Comment by mikeybox** [comment]
Awesome Thanks!
